torch/jit/_isinstance.py

```diff
@@ -0,0 +1,107 @@
+from typing import List, Dict, Tuple, Union, Optional
+
+
+def get_origin(target_type):
+    return getattr(target_type, "__origin__", None)
+
+
+def get_args(target_type):
+    return getattr(target_type, "__args__", None)
+
+
+def check_args_exist(target_type):
+    if target_type is List or target_type is list:
+        raise RuntimeError(
+            "Attempted to use List without a "
+            "contained type. Please add a contained type, e.g. "
+            "List[int]"
+        )
+    elif target_type is Tuple or target_type is tuple:
+        raise RuntimeError(
+            "Attempted to use Tuple without a "
+            "contained type. Please add a contained type, e.g. "
+            "Tuple[int]"
+        )
+    elif target_type is Dict or target_type is dict:
+        raise RuntimeError(
+            "Attempted to use Dict without "
+            "contained types. Please add contained type, e.g. "
+            "Dict[int, int]"
+        )
+    elif target_type is None or target_type is Optional:
```

Comment:
The inclusion of `target_type is None` is bit confusing. `torch.jit.isinstance(x, None)` seems like perfectly valid code to me. It seems to be a bit too tightly coupled to `check_args_exist` and the weird behaviour you noted below where an `Optional` with no contained type has no `__origin__`.

Reply:
`None` is actually not a type in Python and `isinstance(x, None)` is not allowed. Or is there a specific use case you are thinking about?

Reply:
Oh that's true, usually `x is None` is used for that.